js/classes/game_engine.js

```diff
@@ -12,22 +12,24 @@ class GameEngine {
         this.spanish = isSpanish ?? false;
         /** Everything that will be updated and drawn each frame. */
         this.entities = [];
-        /** Is the user pressing S key? */
-        this.down = false;
-        /** Is the user pressing the W key? */
-        this.up = false;
-        /** Is the user pressing the D key? */
-        this.right = false;
-        /** Is the user pressing the A key? */
-        this.left = false;
-        /** Is the user pressing the left shift key? */
-        this.shiftLeft = false;
-        /** Is the user pressing the left X key key? */
-        this.keyX = false;
+        /** Is Chad moving down? */
+        this.movingDown = false;
+        /** Is Chad moving up? */
+        this.movingUp = false;
+        /** Is Chad moving right? */
+        this.movingRight = false;
+        /** Is Chad moving left? */
+        this.movingLeft = false;
+        /** Is Chad jumping? */
+        this.jumping = false;
+        /** Is Chad running? */
+        this.running = false;
+        /** Is Chad dashing mid air? */
+        this.dashing = false;
         /** Is the user pressing the mouse button? */
-        this.mouseDown = false;
+        this.leftClickPressed = false;
         /** Is the user releasing the mouse button? */
-        this.mouseUp = false;
+        this.leftClickReleased = false;
```

Comment:
You have changed the variable names up here, but not down below where the listeners are.

Reply:
My bad for the oversight!